lrtable/src/lib/statetable.rs

```diff
@@ -133,7 +133,8 @@ where
         );
         let maxa = usize::from(grm.tokens_len()) * usize::from(sg.all_states_len());
         let maxg = usize::from(grm.rules_len()) * usize::from(sg.all_states_len());
-        assert!(usize::from(sg.all_states_len()) < usize::from(StIdx::max_value()));
+        // We only have 30 bits to store state IDs
+        assert!(usize::from(sg.all_states_len()) < (u32::max_value() - 4) as usize);
```

Comment:
Can we do `assert!(usize::from(sg.all_states_len()) < usize::from(u32::max_value() - 4))`? I'm keen to avoid `as` whenever possible.

Reply:
Erch, `From<u32> not implemented for usize`.

Reply:
Ah. `assert!(u32::try_from(sg.all_states_len()).unwrap() < u32::max_value() - 4);`?

Reply:
So that would return an error if `sg.all_states_len()` is bigger than `u32`, which then also makes the `assert` fail?

Reply:
Yeah, which is fine.